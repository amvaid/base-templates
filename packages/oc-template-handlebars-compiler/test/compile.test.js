const path = require('path');
const compile = require('../lib/compile.js');

jest.mock('fs-extra');
jest.mock('../lib/compileServer.js');
jest.mock('../lib/compileView.js');
jest.mock('../lib/compileStatics.js');

const fs = require('fs-extra');
fs.writeJSON.mockImplementation((_path, componentPackage, cb) =>
  cb(null, componentPackage)
);
const compileServer = require(
  '../lib/compileServer.js'
).mockImplementation((options, cb) => cb());
const compileView = require(
  '../lib/compileView.js'
).mockImplementation((options, cb) => cb());
const compileStatics = require(
  '../lib/compileStatics.js'
).mockImplementation((options, cb) => cb());

afterEach(() => {
  compileView.mockClear();
  compileServer.mockClear();
  compileStatics.mockClear();
  fs.writeJSON.mockClear();
});

test('Should correctly invoke the various compilers', done => {
  const options = {
    componentPackage: {
      oc: {
        files: {
          template: 'template.hbs',
          data: 'server.js'
        }
      }
    },
    ocPackage: {},
    publishPath: ''
  };
  compile(options, () => {
    expect(compileView).toHaveBeenCalledTimes(1);
    compileView.mock.calls[0][0].componentPackage.oc.date = '';
    expect(compileView.mock.calls).toMatchSnapshot();

    expect(compileServer).toHaveBeenCalledTimes(1);
    compileServer.mock.calls[0][0].componentPackage.oc.date = '';
    expect(compileServer.mock.calls).toMatchSnapshot();

    expect(compileStatics).toHaveBeenCalledTimes(1);
    compileStatics.mock.calls[0][0].componentPackage.oc.date = '';
    expect(compileStatics.mock.calls).toMatchSnapshot();

    expect(fs.writeJSON).toHaveBeenCalledTimes(1);
    expect(fs.writeJSON.mock.calls).toMatchSnapshot();
    done();
  });
});

test('Should correctly invoke the various compilers when no server has been defined', done => {
  const options = {
    componentPackage: {
      oc: {
        files: {
          template: 'template.hbs'
        }
      }
    },
    ocPackage: {},
    publishPath: ''
  };
  compile(options, () => {
    expect(compileView).toHaveBeenCalledTimes(1);
    compileView.mock.calls[0][0].componentPackage.oc.date = '';
    expect(compileView.mock.calls).toMatchSnapshot();

    expect(compileServer).not.toHaveBeenCalled();

    expect(compileStatics).toHaveBeenCalledTimes(1);
    compileStatics.mock.calls[0][0].componentPackage.oc.date = '';
    expect(compileStatics.mock.calls).toMatchSnapshot();

    expect(fs.writeJSON).toHaveBeenCalledTimes(1);
    expect(fs.writeJSON.mock.calls).toMatchSnapshot();
    done();
  });
});
