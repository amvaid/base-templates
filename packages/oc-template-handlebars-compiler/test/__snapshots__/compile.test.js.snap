// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly compile the oc component 1`] = `
Object {
  "description": "",
  "name": "handlebars-component",
  "oc": Object {
    "date": "",
    "files": Object {
      "dataProvider": Object {
        "hashKey": "46fc3cca37f747de97af6e9266895190bc6048ca",
        "src": "server.js",
        "type": "node.js",
      },
      "static": Array [
        "assets",
      ],
      "template": Object {
        "hashKey": "2112a7cce970d8e5eb8a60c5ada1c4d822b22b0e",
        "src": "template.js",
        "type": "handlebars",
      },
    },
    "packaged": true,
    "version": "1.0.0",
  },
  "repository": "",
  "version": "1.0.0",
}
`;

exports[`Should correctly compile the oc component 2`] = `
Array [
  "../../../mocks/handlebars-component/__package/assets/css/style.css",
  "../../../mocks/handlebars-component/__package/assets/img/test.png",
  "../../../mocks/handlebars-component/__package/assets/js/lib.js",
  "../../../mocks/handlebars-component/__package/package.json",
  "../../../mocks/handlebars-component/__package/server.js",
  "../../../mocks/handlebars-component/__package/template.js",
]
`;
